lgc/patch/PatchLoadScalarizer.cpp

```diff
@@ -45,39 +45,68 @@ namespace lgc {
 
 // =====================================================================================================================
 // Define static members (no initializer needed as LLVM only cares about the address of ID, never its value).
-char PatchLoadScalarizer::ID;
+char LegacyPatchLoadScalarizer::ID;
 
 // =====================================================================================================================
 // Pass creator, creates the pass of LLVM patching operations for load scalarizer optimizations.
-FunctionPass *createPatchLoadScalarizer() {
-  return new PatchLoadScalarizer();
+FunctionPass *createLegacyPatchLoadScalarizer() {
+  return new LegacyPatchLoadScalarizer();
 }
 
 // =====================================================================================================================
-PatchLoadScalarizer::PatchLoadScalarizer() : FunctionPass(ID) {
+PatchLoadScalarizer::PatchLoadScalarizer() {
   m_scalarThreshold = 0;
 }
 
+// =====================================================================================================================
+LegacyPatchLoadScalarizer::LegacyPatchLoadScalarizer() : FunctionPass(ID) {
+}
+
 // =====================================================================================================================
 // Get the analysis usage of this pass.
 //
 // @param [out] analysisUsage : The analysis usage.
-void PatchLoadScalarizer::getAnalysisUsage(AnalysisUsage &analysisUsage) const {
+void LegacyPatchLoadScalarizer::getAnalysisUsage(AnalysisUsage &analysisUsage) const {
   analysisUsage.addRequired<LegacyPipelineStateWrapper>();
   analysisUsage.addRequired<LegacyPipelineShaders>();
   analysisUsage.addPreserved<LegacyPipelineShaders>();
 }
 
+// =====================================================================================================================
+// Executes this LLVM pass on the specified LLVM function.
+//
+// @param [in/out] function : Function that we will peephole optimize.
+// @returns : True if the module was modified by the transformation and false otherwise
+bool LegacyPatchLoadScalarizer::runOnFunction(Function &function) {
+  PipelineState *pipelineState = getAnalysis<LegacyPipelineStateWrapper>().getPipelineState(function.getParent());
+  return m_impl.runImpl(function, pipelineState);
+}
+
+// =====================================================================================================================
+// Executes this LLVM pass on the specified LLVM function.
+//
+// @param [in/out] function : Function that we will peephole optimize.
+// @param [in/out] analysisManager : Analysis manager to use for this transformation
+// @returns : The preserved analyses (The Analyses that are still valid after this pass)
```

Comment:
Nit: second "Analyses" -> "analyses" ? (There are more occurrences of this in other files as well.)